proof/crefine/RISCV64/SyscallArgs_C.thy

```diff
@@ -1058,10 +1053,31 @@ lemma index_msgRegisters_less':
 
 lemma index_msgRegisters_less:
   "n < 4 \<Longrightarrow> index msgRegistersC n <s 35"
-  "n < 4 \<Longrightarrow> index msgRegistersC n < 35"
   using index_msgRegisters_less'
   by (simp_all add: word_sless_msb_less)
 
+lemma less_ucast_ucast_less':
+  "x < UCAST('b \<rightarrow> 'a) y \<Longrightarrow> UCAST('a \<rightarrow> 'b) x < y"
```

Comment:
These look more generally useful. Can you please move them up to `Word_Lemmas_Internal`, so I can merge them into the AFP `Word_Lib` over time?

Reply:
The one below it is even nicer. `ucast_up_less_bounded_iff_less_ucast_down'` essentially says that upcast and downcast form a Galois connection.